doc/sphinx/addendum/compiled-interfaces.rst

```diff
@@ -0,0 +1,85 @@
+(i.e., proofs ending with ``Qed`` or ``Admitted``), output a `.vos` files
+instead of a `.vo` file, and to load `.vos` files instead of `.vo` files
+when interpreting `Require` commands.
+
+The option `-vok` indicate |Coq| to check a file completely, to load
+`.vos` files instead of `.vo` files when interpreting `Require` commands,
+and to output an empty `.vok` files upon success instead of writing a
+`.vo` file.
+
+Typical usage
+-------------
+
+Assume a file `foo.v` that depends on two files `bar1.v` and `bar2.v`.
+The command `make foo.requires_vos` will compile `bar1.v` and `bar2.v`
+using the option `-vos` to skip the proofs, producing `bar1.vos` and `bar2.vos`.
+At this point, one is ready to work interactively on the file `foo.v`, even
+though it was never needed to compile the proofs involved in the `bar*.v` files.
+
+Assume a set of files `f1.v ... fn.v` with linear dependencies. The command
+`make vos` enables compiling the statements (i.e. excluding the proofs) in
+all the files, then `make -j vok` enables compiling all the proofs in parallel.
+More generally, calling `make -j vok` directly enables taking advantage of a
+maximal amount of parallelism during the compilation of the set of files.
+This comes at the cost of parsing and typechecking all definitions twice,
+once for the `.vos` file and once for the `.vok` file, but when files contain
+nontrivial proofs, or the files have many linear chains of dependencies, or
+one has many cores available, compilation should be overall faster.
+
+Need for `Proof using`
+----------------------
+
+When a theorem is part of a section, typechecking the statement of this
+theorem might be insufficient for deducing the type of this statement as
+of at the end of the section. Indeed, the proof of the theorem could make
+use of section variables or section hypotheses that do are not mentioned
+in the statement of the theorem.
+
+For this reason, proofs inside section should begin with `Proof using [..].`
+instead of `Proof`, where after the `using` one should provide the list
+of the names of the section variables that are required for the proof
+but are not involved in the typechecking of the statement. For uniformity,
+one is encouraged to use `Proof using` for all proofs, even those that do
+not fall in a section.
+
+If |Coq| is invoked using the `-vos` option, whenever it finds the command
+`Proof.` inside a section, it will compile the proof, that is, refuse to
+skip it, and it will raise a warning. To disable the warning, one may pass
+the flag `-w -proof-without-using-in-section`.
+
+Limitations
+-----------
+
+The interpretation of `Include` for importing modules might not work
+properly with `-vos`, for the same reason as it might not work properly
+with `-quick`. Hopefully, this could be resolved in the near future.
+
+Compiling `.vok` files is not as trustworthy as building `.vo` files,
+because some universe constraints might get lost (?). So, it can be a
```

Comment:
Determine if this is trustworthy or not and then write a more affirmative claim.